### PR TITLE
Cli : Popper search sub-command added

### PR DESCRIPTION
**Pull request by UjjwalAyyangar**
* Added a popper search sub-command that queries github and allows users to list available pipelines of an organization. (Addresses #70 )
* Added support for metadata caching (Addresses #258 )

### CODE REVIEW

File: cli/popper/commands/cmd_search.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    """
+
+    project_root = pu.get_project_root()
+
+    empty_query = False
+
+    if not pipeline_name:  # checks if the query is empty or not
+        empty_query = True
+
+    POPPER_GITHUB_API = ""
```
Comment (ivotron) [inline cli/popper/commands/cmd_search.py]: can we rename it to `POPPER_GITHUB_API_TOKEN`? That would be more closer to the "standard" practice (e.g. what `homebrew` and other cli tools do)

File: docs/protocol/getting_started.md [hunk 2]
```diff
@@ -116,6 +116,19 @@ git push
 
 Go to TravisCI website to see your experiments being executed.
 
+## Integrating with Git-Hub
+
+Some of the popper sub-commands (eg :- popper search) make use of the GitHub
```
Comment (ivotron) [inline docs/protocol/getting_started.md]: change `eg :-` to `e.g.`?

**Comment by UjjwalAyyangar** [comment]
@ivotron - All the requested changes have been made.

File: docs/protocol/getting_started.md [hunk 3]
```diff
@@ -116,6 +116,19 @@ git push
 
 Go to TravisCI website to see your experiments being executed.
 
+## Integrating with GitHub
+
+Some of the popper sub-commands (e.g. :- popper search) make use of the GitHub
+API. Since GitHub only allows up to 60 unauthenticated requests per hour on 
+it's API, some of these sub-commands will fail to give appropriate results on heavy
```
Comment (ivotron) [inline docs/protocol/getting_started.md]: `its` instead of `it's`

File: docs/protocol/getting_started.md [hunk 4]
```diff
@@ -116,6 +116,19 @@ git push
...
+usage. 
+
+To resolve this, we need to :
+  *  Create a GitHub personal access token, as shown [here](https://bit.ly/2rvbeT1).
+  *  Copy the token and set it as an environment variable with the name `POPPER_GITHUB_API_TOKEN` in our computer. 
+ 
+This will allow the Popper to use our access token to make authenticated requests.
```
Comment (ivotron) [inline docs/protocol/getting_started.md]: "allow the `popper` command.

File: docs/protocol/getting_started.md (hunk 3 above)
Comment (UjjwalAyyangar) [reply docs/protocol/getting_started.md]: oops :see_no_evil:

**Comment by ivotron** [comment]
thanks a lot!

**Pull request merged**